gensim/test/test_phrases.py

```diff
@@ -321,9 +321,10 @@ def test_save_load_custom_scorer(self):
     def test_save_load(self):
         """Test saving and loading a Phrases object."""
         with temporary_file("test.pkl") as fpath:
-            bigram = Phrases(self.sentences, min_count=1, threshold=1)
+            bigram = Phrases(self.sentences, min_count=1, threshold=1, connector_words=self.connector_words)
```

Comment:
Why are we changing the behavior of an existing, passing test?

Should we not be adding a new test case here?

Reply:
I wanted to create a separate test, But after reading this whole file gensim/test/test_phrases.py, I found that this test function i.e "test_save_load" should ideally be capturing this bug. Give its name "test_save_load" I assumed that this test was designed to capture the saving & loading of the whole model and this test shouldn't pass if saved connector words are not loaded. Adding it separately will increase the number of lines of code and will increase the test execution time. Another thought can be to remove this test altogether i.e not testing loading & saving of connector words. Let me know I can change the PR.

Reply:
I wanted to create a separate test, But after reading this whole file gensim/test/test_phrases.py, I found that this test function i.e "test_save_load" should ideally be capturing this bug. Give its name "test_save_load" I assumed that this test was designed to capture the saving & loading of the whole model and this test shouldn't pass if saved connector words are not loaded. Adding it separately will increase the number of lines of code and will increase the test execution time. Another thought can be to remove this test altogether i.e not testing loading & saving of connector words. Let me know I can change the PR.

Reply:
> Adding it separately will increase the number of lines of code and will increase the test execution time.

The test is less than 10 lines long, so duplicating them isn't a huge issue. That leaves execution time.

How long does the test take to run?

> Another thought can be to remove this test altogether i.e not testing loading & saving of connector words.

What would be the benefit of removing this test? Isn't it testing a valid case?

Reply:
I couldn't understand your first point. After reading your first point, It seems that you have misunderstood that this bug is loading an old model. This bug occurs when we save and load the model in the same version itself. I mean to say that "test_save_load_old_model" would not describe this bug related test. string like "test_save_load_with_connector_words" can describe this bug
I agree with all of the rest of your points. 
It will take the same time as test  "test_save_load"
Yes its testing a valid case

Reply:
I couldn't understand your first point. After reading your first point, It seems that you have misunderstood that this bug is loading an old model. This bug occurs when we save and load the model in the same version itself. I mean to say that "test_save_load_old_model" would not describe this bug related test. string like "test_save_load_with_connector_words" can describe this bug
I agree with all of the rest of your points.
It will take the same time as test "test_save_load"
Yes its testing a valid case

Reply:
> string like "test_save_load_with_connector_words" can describe this bug

Sure, if that's a more suitable name, please use that instead.

Reply:
Updated PR with new test method and no behaviour changing of an existing, passing test